builder/main.py

```
@@ -862,6 +862,212 @@ def coredump_analysis(target, source, env):
+            break
+    
+    if fs_start is None or fs_size is None:
+        print("Error: No filesystem partition found in partition table")
+        return 1
+    
+    # Check if filesystem is supported
+    # Note: LittleFS can use subtype 0x82 or 0x83
+    # We only support LittleFS extraction, not SPIFFS
+    # The actual filesystem type will be detected when mounting
+    if fs_subtype not in [0x82, 0x83]:
+        print(f"Error: Unsupported filesystem partition type")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)